civihr_employee_portal/src/View/MyDetails_MyRoleView.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace Drupal\civihr_employee_portal\View;
+
+class MyDetails_MyRoleView extends AbstractView {
+
+  /**
+   * @var string
+   */
+  protected static $name = 'MyDetails_MyRole';
+
+  /**
+   * @inheritdoc
+   * To alter specific fields which store dates in format Y-m-d
+   * The dates are converted to Unix Timestamp so they can be used
+   * from the views interface as filters
+   */
+  public function alter($view, $query) {
+    $dateFieldsToTimestamp = [
+      'hrjc_role_hrjc_revision.role_start_date',
+      'hrjc_role_hrjc_revision.role_end_date'
+    ];
+    civihr_employee_portal_views_date_fields_to_timestamp($view, $query, $dateFieldsToTimestamp);
```

Comment:
@mickadoo this function is outside the class because it was designed to be used not only by the current views in the AbstractView class, but also to be used through functional programming from any other view, because it fixes a generic problem for date filters which can be applied to other views.